packages/babel-plugin-transform-async-generator-functions/test/fixtures/for-await/async-generator-exec/exec.js

```diff
@@ -22,5 +24,7 @@ var output = 0;
 return forEach(genAnswers(), function(val) { output += val.value })
 .then(function () {
   assert.equal(output, 42);
+})
+.catch(function (error) {
```

Comment:
Maybe it does work, I just don't think the logic I wrote in https://github.com/babel/babel/commit/5ea57d5e9cb1c049fba142930ba3acde9cc473c2 accounts for .catch